test/filename-case.mjs

```diff
@@ -79,6 +79,7 @@ test({
 		testCase('src/foo/___foo-bar.js', 'kebabCase'),
 		testCase('src/foo/_FooBar.js', 'pascalCase'),
 		testCase('src/foo/___FooBar.js', 'pascalCase'),
+		testCase("src/foo/$foo.js"),
```

Comment:
Can you also add an invalid case with filename contains `$`?

Reply:
`$` in the filename will never cause the filename to be invalid because this character has nothing to do with casing.

Reply:
https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1628/commits/67737fd0bdcae1457e6572dce38fb6e57fb27dc1?